plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -47,6 +50,31 @@ def template
       end
     end
 
+    # process the annotations and do some string substition
```

Comment:
`expand $ENV and $DEPLOY_GROUP in annotations that start with '/secret/' and verify the secret exists before trying to deploy`